controllers/limitador_status.go

```diff
@@ -108,6 +125,117 @@ func (r *LimitadorReconciler) readyCondition(ctx context.Context, limitadorObj *
 	return cond, nil
 }
 
+func (r *LimitadorReconciler) availableCondition(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) (*metav1.Condition, bool, error) {
+	cond := &metav1.Condition{
+		Type:    limitadorv1alpha1.StatusConditionAvailable,
+		Status:  metav1.ConditionTrue,
+		Reason:  "Available",
+		Message: "Limitador limits synced",
+	}
+
+	reason, err := r.checkLimitadorSynced(ctx, limitadorObj)
+	if err != nil {
+		return nil, true, err
+	}
+	if reason != nil {
+		cond.Status = metav1.ConditionFalse
+		cond.Reason = "LimitadorNotAvailable"
+		cond.Message = *reason
+		return cond, true, nil
+	}
+
+	return cond, false, nil
+}
+
+func (r *LimitadorReconciler) checkLimitadorSynced(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) (*string, error) {
+	// Create Kubernetes clientset.
+	clientset, err := kubernetes.NewForConfig(r.RestConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	podList := &corev1.PodList{}
+	options := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(limitador.Labels(limitadorObj)),
+		Namespace:     limitadorObj.Namespace,
+	}
+	if err := r.Client().List(ctx, podList, options); err != nil {
+		return nil, err
+	}
+
+	if len(podList.Items) == 0 {
+		return nil, errors.New("found 0 pods")
+	}
+
+	// Name of the pod where the function will be executed.
+	podName := podList.Items[0].Name
+
+	// Command to execute your function.
+	command := []string{"cat", fmt.Sprintf("%s/%s", limitador.LimitadorCMMountPath, limitador.LimitadorConfigFileName)}
+
+	req := clientset.CoreV1().
+		RESTClient().
+		Post().
+		Resource("pods").
+		Name(podName).
+		Namespace(limitadorObj.Namespace).
+		SubResource("exec")
+
+	option := &corev1.PodExecOptions{
+		Command:   command,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       true,
+		Container: "limitador",
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+
+	// Create an executor.
+	executor, err := remotecommand.NewSPDYExecutor(r.RestConfig, "POST", req.URL())
+	if err != nil {
+		return nil, err
+	}
+
+	// Create buffers to capture stdout and stderr.
+	var stdout, stderr bytes.Buffer
+
+	// Create a StreamOptions struct.
+	streamOptions := remotecommand.StreamOptions{
+		Stdout: &stdout,
+		Stderr: &stderr,
+	}
+
+	// Execute the function within the pod.
+	err = executor.StreamWithContext(ctx, streamOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if stderr.String() != "" {
+		return nil, errors.New(stderr.String())
+	}
+
+	// Get the config map
+	configmap := corev1.ConfigMap{}
+	if err := r.Client().Get(ctx, client.ObjectKey{Namespace: limitadorObj.Namespace, Name: limitador.LimitsConfigMapName(limitadorObj)}, &configmap); err != nil {
+		return nil, err
+	}
+
+	configmapData := configmap.Data[limitador.LimitadorConfigFileName]
+	// There might be line break differences
+	configmapInPod := strings.ReplaceAll(stdout.String(), "\r\n", "\n")
+	if configmapData != configmapInPod {
```

Comment:
for PoC is ok.. but .... too many things can happen that make that different even if they are equal.

Reply:
Agreed. Though I'll note there'll also likely be a difference for if we want to compare using the `/limits/{namespace}` endpoint as `""` / `''` characters as they can be returned differently as JSON (and then marhshalled back as struct) vs how they are in the config map / limitador CR limits struct for what I've tried initially 🤔

Reply:
Up to you, my take on this is to compare go types (list of structs or maps or whatever) taking into account if sorting is necessary.